File: Recursion/CheckArraySorted/checkSorted.cpp
/*
Problem : Check Array is sorted or not.
*/

#include <iostream>
#include <vector>
using namespace std;

bool isSorted(vector<int>& arr, int index){
    //Base case
    if(index >= arr.size()) return true;

    // 2nd Base Case
    if(arr[index] < arr[index-1]) return false;

    return isSorted(arr,index+1);
}

int main(){

    vector<int> arr = {1,2,3,4,5,6,7};

    if(isSorted(arr,1)){
        cout << "Array is sorted" << endl;
    }
    else{
        cout << "Array is not sorted" << endl;
    }


    return 0;
}

File: Recursion/CheckArraySorted/checkSorted.py
"""
Problem : Check Array is sorted or not.
"""

def isSorted(arr,index):
    # Base Case
    if index >= len(arr):
        return True
    
    # 2nd Base Case
    if arr[index] < arr[index-1]:
        return False
    
    return isSorted(arr,index+1)


arr = [1,2,3,4,5,6,7]

if isSorted(arr,1):
    print("Array is Sorted")
else :
    print("Array is not Sorted")